sdk/azcore/core.go

```diff
@@ -173,13 +179,15 @@ func NullValue(v interface{}) interface{} {
 func IsNullValue(v interface{}) bool {
 	// see if our map has a sentinel object for this *T
 	t := reflect.TypeOf(v)
-	if t.Kind() != reflect.Ptr {
+	if k := t.Kind(); k != reflect.Ptr && k != reflect.Slice && k != reflect.Map {
```

Comment:
We cant have nil slices?

Reply:
We can.  This check is saying "if v isn't a pointer-to-type, and it's not a slice or a map, then it can't be nil so return false".

Reply:
Oh, yeah definitely misread this. Thanks!